Tests/KnitTests/AbstractRegistrationTests.swift

```diff
@@ -73,6 +73,16 @@ final class AbstractRegistrationTests: XCTestCase {
         )
     }
 
+    @MainActor
+    func testOptionalAbstractRegistrations() {
+        let assembler = ModuleAssembler([Assembly3()])
+        let string = assembler.resolver.resolve(String?.self) ?? nil
+        XCTAssertNil(string)
+
+        let int = assembler.resolver.resolve(Optional<Int>.self) ?? nil
```

Comment:
Why are there `?? nil` on the resolution calls?

Reply:
It converts from Int?? to Int?